Content.Server/DeviceNetwork/Systems/SingletonDeviceNetServerSystem.cs

```diff
@@ -0,0 +1,165 @@
+﻿using Content.Server.DeviceNetwork.Components;
+using Content.Server.Medical.CrewMonitoring;
+using Content.Server.Power.Components;
+using Content.Server.Station.Systems;
+using Robust.Shared.Timing;
+
+namespace Content.Server.DeviceNetwork.Systems;
+
+/// <summary>
+/// Keeps one active server entity per station. Activates another available one if the currently active server becomes unavailable
+/// Server in this context means an entity that manages the devicenet packets like the <see cref="Content.Server.Medical.CrewMonitoring.CrewMonitoringServerSystem"/>
+/// </summary>
+public sealed class SingletonDeviceNetServerSystem : EntitySystem
+{
+    [Dependency] private readonly DeviceNetworkSystem _deviceNetworkSystem = default!;
+    [Dependency] private readonly StationSystem _stationSystem = default!;
+
+    private const float UpdateRate = 3f;
+    private float _updateDiff;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<SingletonDeviceNetServerComponent, PowerChangedEvent>(OnPowerChanged);
+    }
+
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
+
+        // check update rate
+        _updateDiff += frameTime;
+        if (_updateDiff < UpdateRate)
+            return;
+        _updateDiff -= UpdateRate;
+
+        var servers = EntityQueryEnumerator<SingletonDeviceNetServerComponent>();
+
+        while (servers.MoveNext(out var id, out var server))
+        {
+            //Make sure the server is disconnected when it becomes unavailable
+            if (server.Available)
+                continue;
+
+            if (server.Active)
+                DisconnectServer(id, server);
+        }
+    }
```

Comment:
This looks like it's just hacking around some other state handling by just querying every 3 seconds if a server is active?

Reply:
Yea. I was able to just remove that